scripts/pre_commit_linter.py

```diff
@@ -673,21 +673,21 @@ def _lint_html_files(all_files):
         total_error_count += error_count
     total_files_checked = len(html_files_to_lint)
     if total_error_count:
-        print '(%s files checked, %s errors found)' % (
+        failed_message = '(%s files checked, %s errors found)' % (
             total_files_checked, total_error_count)
+        print failed_message
+        failed_messages.append(failed_message)
```

Comment:
Hmm. I was thinking about using context managers here (and everywhere) in the script to get all the stdout and then collect all of them. We already have a function in the pre_commit_linter script which helps to switch streams. How about we create a single `StringIO.StringIO()` object and use it throughout the script? In that way, we will be able to summarize the errors easily too.
Also, I suggest combining the existing `Summary of Errors` section in the `_pre_commit_linter()` function with the complete `Summary of Errors` section at the end of all the checks. Thanks!

Reply:
Done